src/workflow.ts

```diff
@@ -357,6 +358,90 @@ export class WorkflowContextImpl extends DBOSContextImpl implements WorkflowCont
     return this.startChildWorkflow(wf, ...args);
   }
 
+  // TODO: ConfiguredInstance support
+  async procedure<R>(proc: StoredProcedure<R>, ...args: unknown[]): Promise<R> {
```

Comment:
It would be useful for debugging purposes to have a way of running the stored procedure code locally instead of through the database. For example, there might be a `useCompiledProc` flag in the config passed to the decorator and if it's set to false, execute the function locally. This seems like it should be doable because allowable stored procedure functions are a subset of allowable transaction function.

Reply:
Agreed. When running in the debugger, we use the WorkflowContextDebug class, which has an [implementation of `procedure` ](https://github.com/dbos-inc/dbos-transact/blob/0b0755e99575d81ad5143822325e6156b8a392b8/src/debugger/debug_workflow.ts#L239) similar to the debug implementation of `transaction`.

Reply:
I might go further and make it an option in the decorator to make developing these functions even easier--you can be 100% sure the logic works locally before putting it in the database.

Reply:
If we do this, there should be a linter rule to warn the developer this is set. Maybe also a check in `cloud deploy`?